sdk/identity/Azure.Identity/src/Credentials/CredentialOptionsMapper.cs

```diff
@@ -0,0 +1,122 @@
+                options.AdditionallyAllowedTenants = (credentialOptions as ISupportsAdditionallyAllowedTenants)?.AdditionallyAllowedTenants;
+                options.AuthorityHost = credentialOptions.AuthorityHost;
+                options.IsUnsafeSupportLoggingEnabled = credentialOptions.IsUnsafeSupportLoggingEnabled;
+                options.IsChainedCredential = credentialOptions.IsChainedCredential;
+            }
+
+            if (fileSystem is not null)
+            {
+                options.AuthenticationRecord = GetAuthenticationRecord(fileSystem);
+            }
+
+            return options;
+        }
+
+        return null;
+    }
+
+    internal static AuthenticationRecord GetAuthenticationRecord(IFileSystemService _fileSystem)
+        {
+            var homeDir = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile);
+
+            var authRecordPathLowerCase = Path.Combine(homeDir, ".azure", "ms-azuretools.vscode-azureresourcegroups", "authRecord.json");
+            var authRecordPathUpperCase = Path.Combine(homeDir, ".Azure", "ms-azuretools.vscode-azureresourcegroups", "authRecord.json");
+
+            var authRecordPath = _fileSystem.FileExists(authRecordPathLowerCase) ? authRecordPathLowerCase :
+                                 _fileSystem.FileExists(authRecordPathUpperCase) ? authRecordPathUpperCase : null;
+
+            if (authRecordPath == null)
+            {
+                return null;
+            }
+
+            try
+            {
+                using var authRecordStream = _fileSystem.GetFileStream(authRecordPath);
+                var authRecord = AuthenticationRecord.Deserialize(authRecordStream);
+                if (authRecord != null && !string.IsNullOrEmpty(authRecord.TenantId) && !string.IsNullOrEmpty(authRecord.HomeAccountId))
+                {
+                    return authRecord;
+                }
+            }
+            catch (Exception) { }
+            return null;
+        }
+
+    /// <summary>
+    /// Creates fallback options when broker options are not available.
+    /// This prevents the constructor from failing and defers the error to GetToken.
+    /// </summary>
+    /// <returns>A minimal InteractiveBrowserCredentialOptions instance.</returns>
+    internal static InteractiveBrowserCredentialOptions CreateFallbackOptions()
+    {
+        return new InteractiveBrowserCredentialOptions();
+    }
+
+    /// <summary>
+    /// Creates fallback options from the provided credential options when broker options are not available.
+    /// </summary>
+    /// <param name="credentialOptions"> The credential options to copy properties from.</param>
+    /// <returns> A minimal InteractiveBrowserCredentialOptions instance with properties copied from the provided credential options.</returns>
+    internal static InteractiveBrowserCredentialOptions CreateFallbackOptionsFromCredentialOptions(TokenCredentialOptions credentialOptions)
```

Comment:
I think we can remove the `CreateFallbackOptions` above if we make `credentialOptions` optional here.